aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/customize/s3/S3ExpressDecorator.kt

```diff
@@ -0,0 +1,195 @@
+    private val runtimeConfig = codegenContext.runtimeConfig
+    private val codegenScope =
+        arrayOf(
+            *preludeScope,
+            "IdentityCacheLocation" to
+                RuntimeType.smithyRuntimeApiClient(runtimeConfig)
+                    .resolve("client::identity::IdentityCacheLocation"),
+            "ProvideCredentials" to
+                configReexport(
+                    AwsRuntimeType.awsCredentialTypes(runtimeConfig)
+                        .resolve("provider::ProvideCredentials"),
+                ),
+            "SharedCredentialsProvider" to
+                configReexport(
+                    AwsRuntimeType.awsCredentialTypes(runtimeConfig)
+                        .resolve("provider::SharedCredentialsProvider"),
+                ),
+            "SharedIdentityResolver" to
+                RuntimeType.smithyRuntimeApiClient(runtimeConfig)
+                    .resolve("client::identity::SharedIdentityResolver"),
+            "S3_EXPRESS_SCHEME_ID" to
+                RuntimeType.forInlineDependency(
+                    InlineAwsDependency.forRustFile("s3_express"),
+                ).resolve("auth::SCHEME_ID"),
+        )
+
+    override fun section(section: ServiceConfig) =
+        writable {
+            when (section) {
+                ServiceConfig.BuilderImpl -> {
+                    rustTemplate(
+                        """
+                        /// Sets the credentials provider for S3 Express One Zone
+                        pub fn express_credentials_provider(mut self, credentials_provider: impl #{ProvideCredentials} + 'static) -> Self {
+                            self.set_express_credentials_provider(#{Some}(#{SharedCredentialsProvider}::new(credentials_provider)));
+                            self
+                        }
+                        """,
+                        *codegenScope,
+                    )
+
+                    rustBlockTemplate(
+                        """
+                        /// Sets the credentials provider for S3 Express One Zone
+                        pub fn set_express_credentials_provider(&mut self, credentials_provider: #{Option}<#{SharedCredentialsProvider}>) -> &mut Self
+                        """,
+                        *codegenScope,
+                    ) {
+                        rustBlockTemplate(
+                            """
+                            if let #{Some}(credentials_provider) = credentials_provider
+                            """,
+                            *codegenScope,
+                        ) {
+                            rustTemplate(
+                                """
+                                self.runtime_components.set_identity_resolver(#{S3_EXPRESS_SCHEME_ID}, credentials_provider);
+                                """,
+                                *codegenScope,
+                            )
+                        }
+                        rust("self")
+                    }
```

Comment:
Nit: this style is less readable than a single large block quote.

Reply:
Addressed in f569231.